### PR TITLE
Transferring props through allowed attributes DS-569

### PR DESCRIPTION
**Pull request by literat**
Fixing only Button and name attribute - https://jira.lmc.cz/browse/DS-569. But it was not enough.

### CODE REVIEW

**Review by adamkudrna — COMMENTED** [review]
Seems correct, but… Is there any difference then between whitelisted attributes like `name` and having them as actual props? Shouldn't we just include `name` and `autocomplete` into components' APIs? What do you think is better?

**Comment by literat** [comment]
> Seems correct, but… Is there any difference then between whitelisted attributes like `name` and having them as actual props? Shouldn't we just include `name` and `autocomplete` into components' APIs? What do you think is better?

Well, I think this is an easier way to define allowed attributes that have no default value. I think there is a difference between component API and attributes of the HTML element. Eg. I was actually hoping that you suggest filling in all available attributes of the button element. With this approach, defining that, will be much easier than having an API definition line for each attribute. Just extend the array.

Also I think that is not a good approach to leave native attributes behind our API. Because every time someone would be using any of the native attributes we will need to add them.

File: packages/web-twig/src/Resources/components/TextFieldBase/TextFieldBase.twig [hunk 1]
```diff
@@ -54,14 +54,15 @@
 {%- set _rootClassNames = [ _rootClassName, _rootDisabledClassName, _rootFluidClassName, _rootValidationStateClassName, _class ] -%}
 {%- set _labelClassNames = [ _labelClassName, _labelHiddenClassName, _labelRequiredClassName ] -%}
 {%- set _mainPropsWithoutId = props | filter((value, prop) => prop != 'id') -%}
+{%- set _allowedAttributes = [ 'autocomplete' ]-%}
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web-twig/src/Resources/components/TextFieldBase/TextFieldBase.twig]: Fixed.

**Review by adamkudrna — APPROVED** [review]
OK, since we only allow to add some attributes, I'm fine with the changes. I'd like to get back to it later though so I'm sure I fully understand the concept.

**Pull request merged**